dhall-docs/src/Dhall/Docs/Core.hs

```diff
@@ -0,0 +1,238 @@
+-- | Contains all the functions that generate documentation
+
+{-# LANGUAGE CPP               #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes       #-}
+{-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
+module Dhall.Docs.Core (generateDocs) where
+
+import Data.Monoid         ((<>))
+import Data.Text           (Text)
+import Data.Void           (Void)
+import Dhall.Docs.Embedded
+import Dhall.Docs.Html
+import Dhall.Docs.Markdown
+import Dhall.Docs.Store
+import Dhall.Parser        (Header (..), ParseError (..), exprAndHeaderFromText)
+import Path                (Abs, Dir, File, Path, (</>))
+import Text.Megaparsec     (ParseErrorBundle (..))
+
+-- QuasiQuotes requires importing `Rel` on ghc 8.6.5
+#if MIN_VERSION_GLASGOW_HASKELL(8,6,5,0)
```

Comment:
I think this solution is dangerous because we don't know for sure that all GHC versions from 8.6.5 onwards require the import and all GHC versions before that don't require the import.  For example, if this change in behavior were actually introduced in GHC 8.6.2 then this would fail to build for that GHC version.

More generally, I think it's a risky solution because we don't know if one behavior or the other was intended by the GHC developers, so there's a risk that the behavior changes again in a future GHC version.

I think the safer solution was the idea of use `{-# OPTIONS_GHC -Wno-unused-imports #-}` that @sjakobi suggested earlier

Reply:
Not a big fan of turning off warnings, but you have a good point though.